noodles-bam/src/record/codec/encoder/sequence.rs

```diff
@@ -7,26 +7,27 @@ pub fn put_sequence<B>(dst: &mut B, read_length: usize, sequence: &Sequence) ->
 where
     B: BufMut,
 {
-    // § 1.4.10 "`SEQ`" (2022-08-22): "If not a '*', the length of the sequence must equal the sum
-    // of lengths of `M`/`I`/`S`/`=`/`X` operations in `CIGAR`."
-    if read_length > 0 && sequence.len() != read_length {
-        return Err(io::Error::new(
+    if sequence.is_empty() {
+        Ok(())
```

Comment:
No need to change the shape of the entire function; this just needs to return early.

Reply:
Yup, no problem.